pandas/tests/base/test_value_counts.py

```diff
@@ -339,3 +339,153 @@ def test_value_counts_object_inference_deprecated():
     exp = dti.value_counts()
     exp.index = exp.index.astype(object)
     tm.assert_series_equal(res, exp)
+
+
+def _vc_make_index(kind: str, periods=5, freq="D"):
+    if kind == "dt":
+        return pd.date_range("2016-01-01", periods=periods, freq=freq)
+    if kind == "td":
+        return pd.timedelta_range(Timedelta(0), periods=periods, freq=freq)
+    raise ValueError("kind must be 'dt' or 'td'")
```

Comment:
Can you just write 1 parametrized test (with `DatetimeIndex` and `TimedeltaIndex` as the hardcoded inputs)? You can also use `tm.assert_index_equal`